fake "invalid cop task execution summaries length" log after #51787

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
mysql> create table test(id int, v1 int, v2 int, v3 int, v4 int, index v4_idx(v4));
Query OK, 0 rows affected (0.02 sec)
                                                                                                                                                                                                                                                                                                 mysql> insert into test values(1,2,3,4,5);                                                                                                                                                                                                                                                       Query OK, 1 row affected (0.00 sec)
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 1 row affected (0.01 sec)
Records: 1  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;                                                                                                                                                                                                                               Query OK, 2 rows affected (0.00 sec)                                                                                                                                                                                                                                                             Records: 2  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 4 rows affected (0.00 sec)
Records: 4  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 8 rows affected (0.01 sec)
Records: 8  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;                                                                                                                                                                                                                               Query OK, 16 rows affected (0.01 sec)
Records: 16  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;                                                                                                                                                                                                                               Query OK, 32 rows affected (0.00 sec)                                                                                                                                                                                                                                                            Records: 32  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;                                                                                                                                                                                                                               Query OK, 64 rows affected (0.00 sec)                                                                                                                                                                                                                                                            Records: 64  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 128 rows affected (0.01 sec)
Records: 128  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 256 rows affected (0.01 sec)
Records: 256  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;                                                                                                                                                                                                                               Query OK, 512 rows affected (0.01 sec)                                                                                                                                                                                                                                                           Records: 512  Duplicates: 0  Warnings: 0
mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 1024 rows affected (0.01 sec)
Records: 1024  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 2048 rows affected (0.01 sec)
Records: 2048  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 4096 rows affected (0.03 sec)
Records: 4096  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 8192 rows affected (0.07 sec)
Records: 8192  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 16384 rows affected (0.10 sec)
Records: 16384  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 32768 rows affected (0.21 sec)
Records: 32768  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 65536 rows affected (0.43 sec)
Records: 65536  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 131072 rows affected (0.83 sec)
Records: 131072  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 262144 rows affected (1.69 sec)
Records: 262144  Duplicates: 0  Warnings: 0

mysql> insert into test select id+1,v1+1,v2+1,v3+1,v4+1 from test;
Query OK, 524288 rows affected (3.58 sec)
Records: 524288  Duplicates: 0  Warnings: 0

mysql> explain analyze select v4+10,v3+10,v2+10 from test use index(v4_idx) where v4 > 10 and v3 < 100 limit 1;
+------------------------------------+---------+---------+-----------+------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+---------+------+
| id                                 | estRows | actRows | task      | access object                | execution info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | operator info                                                                                        | memory  | disk |
+------------------------------------+---------+---------+-----------+------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+---------+------+
| Projection_7                       | 1.00    | 1       | root      |                              | time:874.4µs, loops:2, RU:5.529110, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | plus(test.test.v4, 10)->Column#7, plus(test.test.v3, 10)->Column#8, plus(test.test.v2, 10)->Column#9 | 1.11 KB | N/A  |
| └─Limit_9                          | 1.00    | 1       | root      |                              | time:867.4µs, loops:2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | offset:0, count:1                                                                                    | N/A     | N/A  |
|   └─IndexLookUp_14                 | 1.00    | 1       | root      |                              | time:866µs, loops:1, index_task: {total_time: 786µs, fetch_handle: 596µs, build: 1.62µs, wait: 188.4µs}, table_task: {total_time: 2.24ms, num: 7, concurrency: 5}, next: {wait_index: 346.4µs, wait_table_lookup_build: 34µs, wait_table_lookup_resp: 479.3µs}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |                                                                                                      | 98.0 KB | N/A  |
|     ├─IndexRangeScan_10(Build)     | 1.00    | 704     | cop[tikv] | table:test, index:v4_idx(v4) | time:566.4µs, loops:7, cop_task: {num: 2, max: 282.5µs, min: 267.1µs, avg: 274.8µs, p95: 282.5µs, max_proc_keys: 480, p95_proc_keys: 480, tot_proc: 194.8µs, tot_wait: 41.5µs, copr_cache_hit_ratio: 0.00, build_task_duration: 8.63µs, max_distsql_concurrency: 1}, rpc_info:{Cop:{num_rpc:2, total_time:540.9µs}}, tikv_task:{proc max:0s, min:0s, avg: 0s, p80:0s, p95:0s, iters:7, tasks:2}, scan_detail: {total_process_keys: 704, total_process_keys_size: 32384, total_keys: 706, get_snapshot_time: 16.4µs, rocksdb: {key_skipped_count: 704, block: {cache_hit_count: 3}}}, time_detail: {total_process_time: 194.8µs, total_wait_time: 41.5µs, tikv_wall_time: 324.9µs}                                                                   | range:(10,+inf], keep order:false                                                                    | N/A     | N/A  |
|     └─Limit_13(Probe)              | 1.00    | 3       | cop[tikv] |                              | time:1.9ms, loops:10, cop_task: {num: 3, max: 455.7µs, min: 335.5µs, avg: 378.4µs, p95: 455.7µs, max_proc_keys: 32, p95_proc_keys: 32, tot_proc: 427.2µs, tot_wait: 55.3µs, copr_cache_hit_ratio: 0.00, build_task_duration: 88.4µs, max_distsql_concurrency: 1, max_extra_concurrency: 1}, rpc_info:{Cop:{num_rpc:3, total_time:1.13ms}}, tikv_task:{proc max:1ms, min:1ms, avg: 1ms, p80:1ms, p95:1ms, iters:3, tasks:3}, scan_detail: {total_process_keys: 96, total_process_keys_size: 5088, total_keys: 121, get_snapshot_time: 20.8µs, rocksdb: {key_skipped_count: 49, block: {cache_hit_count: 30}}}, time_detail: {total_process_time: 427.2µs, total_wait_time: 55.3µs, total_kv_read_wall_time: 3ms, tikv_wall_time: 702.2µs}            | offset:0, count:1                                                                                    | N/A     | N/A  |
|       └─Selection_12               | 1.00    | 96      | cop[tikv] |                              | tikv_task:{proc max:1ms, min:1ms, avg: 1ms, p80:1ms, p95:1ms, iters:3, tasks:3}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | lt(test.test.v3, 100)                                                                                | N/A     | N/A  |
|         └─TableRowIDScan_11        | 1.00    | 96      | cop[tikv] | table:test                   | tikv_task:{proc max:1ms, min:1ms, avg: 1ms, p80:1ms, p95:1ms, iters:3, tasks:3}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | keep order:false                                                                                     | N/A     | N/A  |
+------------------------------------+---------+---------+-----------+------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+---------+------+
7 rows in set (0.00 sec)
```
### 2. What did you expect to see? (Required)
No "invalid cop task execution summaries length" in log
### 3. What did you see instead (Required)
```
[2025/11/26 11:06:07.728 +08:00] [ERROR] [select_result.go:598] ["invalid cop task execution summaries length"] [expected=3] [received=0]
```
### 4. What is your TiDB version? (Required)
master
<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/distsql/select_result.go
+++ b/pkg/distsql/select_result.go
@@ -443,7 +443,7 @@ func (r *selectResult) fetchRespWithIntermediateResults(ctx context.Context, int
 		if ok {
 			copStats := hasStats.GetCopRuntimeStats()
 			if copStats != nil {
-				if err := r.updateCopRuntimeStats(ctx, copStats, resultSubset.RespTime()); err != nil {
+				if err := r.updateCopRuntimeStats(ctx, copStats, resultSubset.RespTime(), false); err != nil {
 					return err
 				}
 				r.ctx.ExecDetails.MergeCopExecDetails(&copStats.CopExecDetails, duration)
@@ -602,7 +602,7 @@ func recordExecutionSummariesForTiFlashTasks(runtimeStatsColl *execdetails.Runti
 	FillDummySummariesForTiFlashTasks(runtimeStatsColl, storeType, allPlanIDs, recordedPlanIDs)
 }
 
-func (r *selectResult) updateCopRuntimeStats(ctx context.Context, copStats *copr.CopRuntimeStats, respTime time.Duration) (err error) {
+func (r *selectResult) updateCopRuntimeStats(ctx context.Context, copStats *copr.CopRuntimeStats, respTime time.Duration, forUnconsumedStats bool) (err error) {
 	callee := copStats.CalleeAddress
 	if r.rootPlanID <= 0 || r.ctx.RuntimeStatsColl == nil || (callee == "" && (copStats.ReqStats == nil || copStats.ReqStats.GetRPCStatsCount() == 0)) {
 		return
@@ -671,7 +671,7 @@ func (r *selectResult) updateCopRuntimeStats(ctx context.Context, copStats *copr
 			// for TiFlash streaming call(BatchCop and MPP), it is by design that only the last response will
 			// carry the execution summaries, so it is ok if some responses have no execution summaries, should
 			// not trigger an error log in this case.
-			if !(r.storeType == kv.TiFlash && len(r.selectResp.GetExecutionSummaries()) == 0) {
+			if !forUnconsumedStats && !(r.storeType == kv.TiFlash && len(r.selectResp.GetExecutionSummaries()) == 0) {
 				logutil.Logger(ctx).Warn("invalid cop task execution summaries length",
 					zap.Int("expected", len(r.copPlanIDs)),
 					zap.Int("received", len(r.selectResp.GetExecutionSummaries())))
@@ -734,7 +734,7 @@ func (r *selectResult) close() error {
 		if unconsumed, ok := r.resp.(copr.HasUnconsumedCopRuntimeStats); ok && unconsumed != nil {
 			unconsumedCopStats := unconsumed.CollectUnconsumedCopRuntimeStats()
 			for _, copStats := range unconsumedCopStats {
-				_ = r.updateCopRuntimeStats(context.Background(), copStats, time.Duration(0))
+				_ = r.updateCopRuntimeStats(context.Background(), copStats, time.Duration(0), true)
 				r.ctx.ExecDetails.MergeCopExecDetails(&copStats.CopExecDetails, 0)
 			}
 		}

--- a/pkg/distsql/select_result_test.go
+++ b/pkg/distsql/select_result_test.go
@@ -43,7 +43,7 @@ func TestUpdateCopRuntimeStats(t *testing.T) {
 	sr.rootPlanID = 1234
 	backOffSleep := make(map[string]time.Duration, 1)
 	backOffSleep["RegionMiss"] = time.Duration(100)
-	sr.updateCopRuntimeStats(context.Background(), &copr.CopRuntimeStats{CopExecDetails: execdetails.CopExecDetails{CalleeAddress: "a", BackoffSleep: backOffSleep}}, 0)
+	sr.updateCopRuntimeStats(context.Background(), &copr.CopRuntimeStats{CopExecDetails: execdetails.CopExecDetails{CalleeAddress: "a", BackoffSleep: backOffSleep}}, 0, false)
 	// RuntimeStatsColl is nil, so the update doesn't take efffect
 	require.Equal(t, sr.stats.backoffSleep["RegionMiss"], time.Duration(0))
 
@@ -60,7 +60,7 @@ func TestUpdateCopRuntimeStats(t *testing.T) {
 	require.NotEqual(t, len(sr.copPlanIDs), len(sr.selectResp.GetExecutionSummaries()))
 
 	backOffSleep["RegionMiss"] = time.Duration(200)
-	sr.updateCopRuntimeStats(context.Background(), &copr.CopRuntimeStats{CopExecDetails: execdetails.CopExecDetails{CalleeAddress: "callee", BackoffSleep: backOffSleep}}, 0)
+	sr.updateCopRuntimeStats(context.Background(), &copr.CopRuntimeStats{CopExecDetails: execdetails.CopExecDetails{CalleeAddress: "callee", BackoffSleep: backOffSleep}}, 0, false)
 	require.False(t, ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.ExistsCopStats(1234))
 	require.Equal(t, sr.stats.backoffSleep["RegionMiss"], time.Duration(200))
 
@@ -69,7 +69,7 @@ func TestUpdateCopRuntimeStats(t *testing.T) {
 	require.Equal(t, len(sr.copPlanIDs), len(sr.selectResp.GetExecutionSummaries()))
 
 	backOffSleep["RegionMiss"] = time.Duration(300)
-	sr.updateCopRuntimeStats(context.Background(), &copr.CopRuntimeStats{CopExecDetails: execdetails.CopExecDetails{CalleeAddress: "callee", BackoffSleep: backOffSleep}}, 0)
+	sr.updateCopRuntimeStats(context.Background(), &copr.CopRuntimeStats{CopExecDetails: execdetails.CopExecDetails{CalleeAddress: "callee", BackoffSleep: backOffSleep}}, 0, false)
 	require.Equal(t, "tikv_task:{time:1ns, loops:1}", ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.GetCopStats(1234).String())
 	require.Equal(t, sr.stats.backoffSleep["RegionMiss"], time.Duration(500))
 }

